### PR TITLE
English documentation about wallabag API

### PR DESCRIPTION
**Pull request by nicosomb**
| Q             | A
| ------------- | ---
| Documentation | yes
| Translation   | yes
| License       | MIT

### CODE REVIEW

File: docs/en/developer/api.rst [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+* all the API methods are documented here http://localhost:8000/api/doc
+
+Create new API client
+---------------------
+
+In your wallabag account, you can create a new API client at this URL ``http://localhost:8000/developer/client/create``.
+
+Just give the redirect URL of your application and create your client.
+
+You have information like this:
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: You get informations like this

File: docs/en/developer/api.rst [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+* wallabag freshly installed on http://localhost:8000
+* ``httpie`` installed on your computer (`see project website <https://github.com/jkbrzt/httpie>`__)
+* all the API methods are documented here http://localhost:8000/api/doc
+
+Create new API client
+---------------------
+
+In your wallabag account, you can create a new API client at this URL ``http://localhost:8000/developer/client/create``.
+
+Just give the redirect URL of your application and create your client.
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: If your application is a desktop application, put whatever urls suits you.

File: docs/en/developer/api.rst [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+    Client secret:
+
+    636ocbqo978ckw0gsw4gcwwocg8044sco0w8w84cws48ggogs4
+
+
+Create refresh token
+--------------------
+
+For each API call, you need a token. Let's create it via this command (replace ``client_id``, ``client_secret``, ``username`` and ``password`` values):
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: you need --> you'll need
via this command --> with this command
values --> with their values

File: docs/en/developer/api.rst [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+::
+
+    http POST http://localhost:8000/oauth/v2/token \
+        grant_type=password \
+        client_id=1_3o53gl30vhgk0c8ks4cocww08o84448osgo40wgw4gwkoo8skc \
+        client_secret=636ocbqo978ckw0gsw4gcwwocg8044sco0w8w84cws48ggogs4 \
+        username=wallabag \
+        password=wallabag
+
+You'll have this return:
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: this in return

File: docs/en/developer/api.rst [hunk 5]
```diff
@@ -0,0 +1,235 @@
+API documentation
+=================
+
+Thanks to this documentation, we'll see how to interact with the wallabag API.
+
+Requirements
+------------
+
+* wallabag freshly installed on http://localhost:8000
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: freshly or not :D

File: docs/en/developer/api.rst [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+* ``httpie`` installed on your computer (`see project website <https://github.com/jkbrzt/httpie>`__)
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: (Note that you can also adapt the commands using curl or wget)

File: docs/en/developer/api.rst [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+* all the API methods are documented here http://localhost:8000/api/doc
+
+Create new API client
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: Creating

File: docs/en/developer/api.rst [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+    Client ID:
+
+    1_3o53gl30vhgk0c8ks4cocww08o84448osgo40wgw4gwkoo8skc
+
+    Client secret:
+
+    636ocbqo978ckw0gsw4gcwwocg8044sco0w8w84cws48ggogs4
+
+
+Create refresh token
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: Obtaining a refresh token

File: docs/en/developer/api.rst [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+        "updated_at": "2016-04-05T09:07:54+0000",
+        "url": "http://www.numerama.com/tech/160115-le-pocket-libre-wallabag-fait-le-plein-de-fonctionnalites.html",
+        "user_email": "",
+        "user_id": 1,
+        "user_name": "wallabag"
+    }
+
+Now, if you execute the previous command (see **Get existing entries**), you'll have data.
+
+Delete an entry
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: Deleting

File: docs/en/developer/api.rst [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+        },
+        "limit": 30,
+        "page": 1,
+        "pages": 1,
+        "total": 0
+    }
+
+The ``items`` array is empty.
+
+Add first entry
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: Adding your first...

File: docs/en/developer/api.rst [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+        "access_token": "ZGJmNTA2MDdmYTdmNWFiZjcxOWY3MWYyYzkyZDdlNWIzOTU4NWY3NTU1MDFjOTdhMTk2MGI3YjY1ZmI2NzM5MA",
+        "expires_in": 3600,
+        "refresh_token": "OTNlZGE5OTJjNWQwYzc2NDI5ZGE5MDg3ZTNjNmNkYTY0ZWZhZDVhNDBkZTc1ZTNiMmQ0MjQ0OThlNTFjNTQyMQ",
+        "scope": null,
+        "token_type": "bearer"
+    }
+
+We'll work with the ``access_token`` value in our next calls.
+
+Get existing entries
```
Comment (tcitworld) [inline docs/en/developer/api.rst]: Getting

File: docs/fr/developer/api.rst [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+        },
+        "limit": 30,
+        "page": 1,
+        "pages": 1,
+        "total": 0
+    }
+
+Le tableau ``items`` est vide.
+
+Créer le premier article
```
Comment (tcitworld) [inline docs/fr/developer/api.rst]: votre

**Comment by tcitworld** [comment]
Last corrections. We're good. :-)

**Comment by nicosomb** [comment]
We are good!

File: src/Wallabag/CoreBundle/Resources/views/themes/material/Developer/index.html.twig [hunk 13]
```diff
@@ -14,7 +14,8 @@
 
                 <ul>
                     <li><a href="{{ path('developer_howto_firstapp') }}">{{ 'developer.how_to_first_app'|trans }}</a></li>
-                    <li><a href="{{ path('nelmio_api_doc_index') }}">{{ 'developer.full_documentation'|trans }}</a></li>
+                    <li><a href="http://doc.wallabag.org/en/v2-documentation-api/developer/api.html">{{ 'developer.full_documentation'|trans }}</a></li>
+                    <li><a href="{{ path('nelmio_api_doc_index') }}">{{ 'developer.list_methods'|trans }}</a></li>
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/material/Developer/index.html.twig]: Why don't you change that text in baggy too?
Also, if you do not plan to use `developer.full_documentation`, you can remove it from the translations files.

**Comment by j0k3r** [comment]
This PR is a good candidate for the merge & squash button :rocket:

File: src/Wallabag/CoreBundle/Resources/views/themes/material/Developer/index.html.twig (hunk 13 above)
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Resources/views/themes/material/Developer/index.html.twig]: `full_documentation` is still used.

I forgot for baggy theme, I'm fixing.

**Pull request merged**